### ISSUE TITLE
Issue with 'notify' parameter of collaborate method

### ISSUE THREAD

**Issue by lyugaloki** [box/box-java-sdk#544]
Hi there,

I am trying to turn off email notification for collaborate invite,
but the invited person still receives email notification.

Below is the code I am using to do this:

```
BoxFolder testFolder = new BoxFolder(api, "FolderID");
testFolder.collaborate("Email Address", BoxCollaboration.Role.EDITOR, false, false);
```

Although I am setting the 3rd parameter ('notify') false,
the invited person still receives an email.

This seems like a glitch to me. Would you be able to fix this?
Thanks!

**Comment by mattwiller** [comment]
Hi @lyugaloki — this does look like a bug in the SDK.  I'll get started on a fix right away!

**Comment by StevenRieger** [comment]
I'm having the same issue.  How do I get the fix?

**Comment by mattwiller** [comment]
@StevenRieger The fix landed in v2.12.0, so as long as you're using at least that version of the SDK you should be able to correctly pass the `notify` query parameter.  That commit includes a passing unit test that appears to show that the parameter is being correctly sent through to the API, and I haven't been able to reproduce the issue myself with the current version of the Java SDK.  When I run the following code, the collaboration is created but no email is sent:

```java
String folderID = "12345";
String collaboratorLogin = "invite.email@example.com";
BoxCollaboration.Role collaboratorRole = BoxCollaboration.Role.VIEWER;

BoxFolder folder = new BoxFolder(api, folderID);
BoxCollaboration.Info collabInfo = folder.collaborate(collaboratorLogin, collaboratorRole, false, null);
```

So, we may need to do a little troubleshooting — here are some questions that might help us figure out what the issue is:

1. First, are you on a version of the SDK after v2.12.0?  If not, upgrading might help.
2. Is there a significant difference between the code you're using and the code I pasted in above?  If so, could you try using something like the sample code and see if that helps?
3. What sort of use is initiating the collaboration?  I'm not sure if there's any meaningful different here between a normal Managed User vs. an Enterprise Admin vs. a Service Account or App User, but it might be helpful to know just in case.
4. If none of the above is helpful, would you mind sending your API key to us at sdks@box.com?  We can take a look and see if there's anything we can find in the server logs that might tell us more about why this isn't working for you.

**Comment by StevenRieger** [comment]
Strange, my maven pom.xml said version 2.5.0 so I thought I'd have that
fix.

I changed it to 2.7.0 and sure enough it fixed it.

Regards,

STeveR
                                                                                        
   ____________________________________________________________________________________ 
                                                                                        
   STEVEN RIEGER CERTIFIED IBM DEVELOPER / ADMINISTRATOR                                
                                                                                        
   DIRECTOR INFORMATION TECHNOLOGIES                                                    
                                                                                        
   Lieberman Management Services, Corporate Office                                      
                                                                                        
   25 Northwest Point Blvd, Suite 330, Elk Grove Village, IL 60007                      
                                                                                        
   Direct / Fax (847) 777.7094 Mobile (847) 219.3961                                    
                                                                                        
                                                                                        
                                                                                        
                                                                                        
                                                                                        





From:	Matt Willer <notifications@github.com>
To:	box/box-java-sdk <box-java-sdk@noreply.github.com>
Cc:	Steven Rieger <srieger@lmsnet.com>, Mention
            <mention@noreply.github.com>
Date:	02/08/2019 05:00 PM
Subject:	Re: [box/box-java-sdk] Issue with 'notify' parameter of
            collaborate method (#544)





@StevenRieger The fix landed in v2.12.0, so as long as you're using at
least that version of the SDK you should be able to correctly pass the
notify query parameter. That commit includes a passing unit test that
appears to show that the parameter is being correctly sent through to the
API, and I haven't been able to reproduce the issue myself with the current
version of the Java SDK. When I run the following code, the collaboration
is created but no email is sent:


String folderID = "12345";

String collaboratorLogin = "invite.email@example.com";

BoxCollaboration.Role collaboratorRole = BoxCollaboration.Role.VIEWER;



BoxFolder folder = new BoxFolder(api, folderID);

BoxCollaboration.Info collabInfo = folder.collaborate(collaboratorLogin,
collaboratorRole, false, null);


So, we may need to do a little troubleshooting — here are some questions
that might help us figure out what the issue is:
   1.	First, are you on a version of the SDK after v2.12.0? If not,
      upgrading might help.
   2.	Is there a significant difference between the code you're using and
      the code I pasted in above? If so, could you try using something like
      the sample code and see if that helps?
   3.	What sort of use is initiating the collaboration? I'm not sure if
      there's any meaningful different here between a normal Managed User
      vs. an Enterprise Admin vs. a Service Account or App User, but it
      might be helpful to know just in case.
   4.	If none of the above is helpful, would you mind sending your API key
      to us at sdks@box.com? We can take a look and see if there's anything
      we can find in the server logs that might tell us more about why this
      isn't working for you.


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.





Please consider the environment before you print this message.  The information contained in this e-mail message may be privileged, confidential, and protected from disclosure. If you are not the intended recipient, any further disclosure or use, dissemination, distribution, or copying of this message or any attachment is strictly prohibited. If you think that you have received this e-mail message in error, please delete it and notify Lieberman Management Services, Inc. either by phone at (847)459-0000 or email at  service@lmsnet.com  and destroy all copies of the original message.  http://www.liebermanmanagement.com
